src/Tokens/TokensList/TokenBalance/TokenBalance.js

```diff
@@ -3,19 +3,36 @@
 //
 // SPDX-License-Identifier: MIT
 
-import React, { PureComponent } from 'react';
+import React, { Component } from 'react';
+import abi from '@parity/shared/lib/contracts/abi/eip20';
+import { defaultAccount$, makeContract$ } from '@parity/light.js';
+import { Link } from 'react-router-dom';
+import { map, switchMap } from 'rxjs/operators';
 import PropTypes from 'prop-types';
 
 import BalanceLayout from '../BalanceLayout';
+import light from '../../../hoc';
 
-class TokenBalance extends PureComponent {
+@light({
+  balance: ({ token: { address, decimals } }) =>
+    defaultAccount$().pipe(
+      switchMap(defaultAccount =>
+        makeContract$(address, abi).balanceOf$(defaultAccount)
+      ),
+      map(value => +value.div(10 ** decimals))
```

Comment:
cc @jacogr That's how an end-developer would use the `@parity/light.js` library.

Pros:
- If you know rxjs, then it's natural

Cons:
- If you don't know rxjs, then it's a headache

PS: could have done `const contract = makeContract$(address, abi)` in a parent and pass it as props. But `makeContract()` is memoized so it doesn't create a new contract instance on each `defaultAccount$()` event.